### PR TITLE
etcher: 1.7.9 -> 1.18.4

### PR DESCRIPTION
**Pull request by Jared-Prime**
###### Description of changes

This is my first PR to the nixpkgs project; happy to learn the ins-and-outs of contributing.

I noticed that [`etcher`](https://etcher.io/), a tool for flashing USB drives, would not build with the default nix package configuration due to an outdated [`electron`](https://electronjs.org/) dependency. My first pass is to get a passing build for the latest version of `etcher` targeting `x86_64-linux`. I'm focusing on an Ubuntu development environment at the moment. If all goes well, I'm happy to help contribute configurations for other platforms as well.

Changes made:

* Add jared-davis to maintainer list
* Create separate configuration for etcher 1.18

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wegank** [comment]
> I noticed that [`etcher`](https://etcher.io/), a tool for flashing USB drives, would not build with the default nix package configuration due to an outdated [`electron`](https://electronjs.org/) dependency.

You don't need to create a separate derivation, just replace the old one with your own.

File: pkgs/tools/misc/etcher/1_18.nix [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    makeWrapper ${electron}/bin/electron $out/bin/${pname} \
+      --add-flags $out/share/${pname}/resources/app \
+      --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath [ gcc-unwrapped.lib ]}"
+  '';
+
+  meta = with lib; {
+    description = "Flash OS images to SD cards and USB drives, safely and easily";
+    homepage = "https://etcher.io/";
+    license = licenses.asl20;
+    maintainers = [ maintainers.shou maintainers.jared-davis ];
```
Comment (felschr) [inline pkgs/tools/misc/etcher/1_18.nix:78]:
```suggestion
    maintainers = with maintainers; [ shou jared-davis ];
```

File: pkgs/tools/misc/etcher/1_18.nix [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+  sha256 = {
+    "x86_64-linux" = "e59563b5424a36bfbc6095466da219228eff6d9e";
+  }."${system}" or throwSystem;
+
+  arch = {
+    "x86_64-linux" = "amd64";
+  }."${system}" or throwSystem;
+
+in
```
Comment (felschr) [inline pkgs/tools/misc/etcher/1_18.nix:24]: Since only one platform remains now, this can also be inlined in `src` below.

**Review by felschr — CHANGES_REQUESTED** [review]
Instead of creating a new package we should just update the existing one.
Newer versions of etcher apparently don't distribute i686-linux builds anymore, but this change is fine for nixos-unstable.

Also, please take a look at [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md), especially regarding commit messages. You can also take a look at previous commit messages in nixpkgs for reference.

If you change this PR to update the existing package this isn't relevant anymore but in case of new packages you also need to add them to [`pkgs/top-level/all-packages`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/all-packages.nix).

**Comment by wegank** [comment]
It seems that Etcher displays a blank screen whatever the version of Electron is on my machine. @Jared-Prime does this happen on your side?

**Comment by felschr** [comment]
Result of `nixpkgs-review pr 237626` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>etcher</li>
  </ul>
</details>

**Comment by felschr** [comment]
> It seems that Etcher displays a blank screen whatever the version of Electron is on my machine. @Jared-Prime does this happen on your side?

I get the same blank screen and I see the following output in the console:
```console
DRI driver not from this Mesa build ('23.0.3' vs '23.1.1')
failed to bind extensions
```

I assume that's because I'm on NixOS 23.05, though.

**Comment by wegank** [comment]
> It seems that Etcher displays a blank screen whatever the version of Electron is on my machine. @Jared-Prime does this happen on your side?

Fixed by adding udev to runtime dependencies.

**Pull request closed without merging**